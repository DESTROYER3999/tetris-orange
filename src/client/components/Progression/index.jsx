import { connect } from "react-redux";
import ProgressionSub from "./subcomponent";

const mapStateToProps = state => ({
  score: state.room.score,
  level: state.room.level
});

const Progression = connect(
  mapStateToProps,
  null
)(ProgressionSub);

export default Progression;
