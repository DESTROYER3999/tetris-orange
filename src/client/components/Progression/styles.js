import styled from "styled-components";

export const StyledContainer = styled.div`
  width: 100%;
  // padding-top: 1vh;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const StyledLabel = styled.div`
  font-size: 3vh;
  padding: 1vh;
`;

export const StyledScoreContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.3);

  div {
    width: 100%;
    display: flex;
    justify-content: center;
    padding: 5px;
  }
  .score-title {
    p {
      font-size: 3vh;
    }
  }
  .score-value {
    background-color: rgba(0, 0, 0, 0.2);
    // border-top: 1px solid white;
    // border-bottom: 1px solid white;
    span {
      font-size: 2.5vh;
    }
  }
`;
export const StyledLevelContainer = styled.div`
  padding: 4px 0 4px 0;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-top: 3px solid white;
  border-bottom: 3px solid white;
  background-color: rgba(0, 0, 0, 0.5);
  span {
    font-size: 1.8vh;
  }
`;
