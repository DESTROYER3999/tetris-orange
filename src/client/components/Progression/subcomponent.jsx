import React from "react";
import {
  StyledContainer,
  StyledLevelContainer,
  StyledScoreContainer
} from "./styles";

const ProgressionSub = ({ score, level }) => {
  return (
    <StyledContainer id="Progression-container">
      <StyledScoreContainer id="Score-container">
        <div className="score-title">
          <p>Score</p>
        </div>
        <div className="score-value">
          <span>{score}</span>
        </div>
      </StyledScoreContainer>
      <StyledLevelContainer id="Level-container">
        <span>Level {level}</span>
      </StyledLevelContainer>
    </StyledContainer>
  );
};

ProgressionSub.propTypes = {};

export default ProgressionSub;
