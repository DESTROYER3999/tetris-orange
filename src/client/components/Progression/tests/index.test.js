import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Progression from "../index";
import socket from "../../../services/socket-api";

const mockStore = configureStore();
const initialState = {
  room: {
    score: 45230,
    level: 23
  }
};

const store = mockStore(initialState);

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<Progression />", () => {
  test("Progression rendering with store without crashing", () => {
    const wrapper = shallow(<Progression store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
