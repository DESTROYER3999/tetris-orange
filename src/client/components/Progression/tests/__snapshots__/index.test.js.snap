// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progression /> Progression rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <ProgressionSub
    dispatch={[Function]}
    level={23}
    score={45230}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
